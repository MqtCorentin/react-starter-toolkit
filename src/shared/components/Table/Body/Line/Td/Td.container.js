import { node } from 'prop-types';
import HelpHover from 'shared/components/HelpInfo';
import Td from './Td';

const TdContainer = ({ children, label, hover, classModifier }) => (
  <Td classModifier={classModifier}>
    <HelpHover content={hover} classModifier="content">
      {label}
      {children}
    </HelpHover>
  </Td>
);

TdContainer.propTypes = {
  children: node,
};

TdContainer.defaultProps = {
  children: null,
};

export default TdContainer;
