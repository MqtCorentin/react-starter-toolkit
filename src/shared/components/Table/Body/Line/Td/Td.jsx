import React from 'react';
import PropTypes from 'prop-types';
import { Table } from '@axa-fr/react-toolkit-all';

const Td = ({ children, ...restTd }) => (
  <Table.Td {...restTd} >{children}</Table.Td>
);

Td.propTypes = {
  children: PropTypes.node,
};

Td.defaultProps = {
  children: null
};

export default Td;
