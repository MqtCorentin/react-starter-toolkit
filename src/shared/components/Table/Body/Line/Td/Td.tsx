import { ReactNode } from 'react';
import HelpHover, { THelpInfo } from 'shared/components/HelpInfo';
import { default as TableTk } from '@axa-fr/react-toolkit-table';

export type TTdContainer = Omit<THelpInfo, 'children'> & {
  children?: ReactNode;
  label?: ReactNode;
  classModifier?: string;
  hover?: ReactNode;
  TdCmpt?: typeof TableTk.Td;
  HelpHoverCmpt?: typeof HelpHover;
};

const TdContainer = ({ children, label, hover, classModifier, TdCmpt = TableTk.Td, HelpHoverCmpt = HelpHover }: TTdContainer) => (
  <TdCmpt classModifier={classModifier}>
    <HelpHoverCmpt content={hover} classModifier="content">
      {label}
      {children}
    </HelpHoverCmpt>
  </TdCmpt>
);

export default TdContainer;
