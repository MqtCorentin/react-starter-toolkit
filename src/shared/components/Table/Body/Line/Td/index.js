export { default } from './Td.container';
