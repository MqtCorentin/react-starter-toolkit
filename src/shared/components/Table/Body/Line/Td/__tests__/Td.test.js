import { render } from '@testing-library/react';
import Td from '../Td';

const defaultProps = {
  children: null,
};
const container = document.createElement('tr');

const renderWithContainer = Component =>
  render(<Component />, {
    container: document.body.appendChild(container),
  });

describe('Td', () => {
  it('Render <Td/>', () => {
    const { asFragment } = renderWithContainer(() => <Td {...defaultProps}>child td</Td>);
    expect(asFragment()).toMatchSnapshot();
  });
});
