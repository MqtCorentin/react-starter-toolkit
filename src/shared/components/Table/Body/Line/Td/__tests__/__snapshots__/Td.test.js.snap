// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Td Render <Td/> 1`] = `
<DocumentFragment>
  child td
</DocumentFragment>
`;
